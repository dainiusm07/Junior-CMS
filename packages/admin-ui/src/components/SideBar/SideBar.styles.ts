import { makeStyles } from '@material-ui/core';

const drawerWidth = 240;

export default makeStyles((theme) => ({
  drawer: {
    zIndex: theme.zIndex.drawer,
    width: drawerWidth,
    flexShrink: 0,
    boxShadow: theme.shadows[3],
  },
  drawerPaper: {
    padding: theme.spacing(2),
    width: drawerWidth,
    position: 'relative',
  },
}));
