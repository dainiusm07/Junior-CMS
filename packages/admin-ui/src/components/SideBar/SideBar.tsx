import { Drawer, Typography } from '@material-ui/core';
import React from 'react';

import useStyles from './SideBar.styles';

const SideBar: React.FC = () => {
  const classes = useStyles();

  return (
    <Drawer
      variant="permanent"
      className={classes.drawer}
      classes={{
        paper: classes.drawerPaper,
      }}
    >
      <Typography>SideBar</Typography>
    </Drawer>
  );
};

export default SideBar;
