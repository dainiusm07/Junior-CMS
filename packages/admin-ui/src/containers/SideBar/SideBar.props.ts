export type SideBarProps = {
  open: boolean;
  handleClose: () => void;
};
