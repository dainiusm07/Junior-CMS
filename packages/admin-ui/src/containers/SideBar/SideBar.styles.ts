import { makeStyles } from '@material-ui/core';

const drawerWidth = 240;

export default makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  drawerPaper: {
    padding: theme.spacing(0, 2, 2),
    width: drawerWidth,
    position: 'relative',
  },
}));
