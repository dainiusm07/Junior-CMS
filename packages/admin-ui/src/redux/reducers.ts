import { combineReducers } from 'redux';

import { GlobalState } from './types';
import UiReducer from './Ui/Ui.reducers';
import UserReducer from './User/User.reducers';

const reducers = combineReducers<GlobalState>({
  user: UserReducer,
  ui: UiReducer,
});

export default reducers;
