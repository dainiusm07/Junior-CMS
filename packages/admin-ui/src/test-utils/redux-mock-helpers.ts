export const mockedDispatch = jest.fn();
export const mockedUseSelector = jest.fn();

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedDispatch,
  useSelector: mockedUseSelector,
}));
