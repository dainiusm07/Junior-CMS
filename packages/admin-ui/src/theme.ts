import { createMuiTheme } from '@material-ui/core';

const theme = createMuiTheme({
  palette: {
    primary: { main: '#736CED' },
    background: { paper: '#222b36', default: '#171c24' },
    text: { primary: '#ffffff', secondary: '#919eab' },
  },
  shape: { borderRadius: 15 },
  overrides: {
    MuiOutlinedInput: {
      input: {
        padding: '16.5px 14px',
      },
    },
  },
});

export default theme;
