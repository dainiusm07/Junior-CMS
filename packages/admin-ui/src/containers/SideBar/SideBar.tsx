import { Drawer, Typography, useMediaQuery, useTheme } from '@material-ui/core';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { closeSideBar } from '../../redux/Ui/Ui.actions';
import { sideBarOpenedSelector } from '../../redux/Ui/Ui.selectors';
import { SideBarProps } from './SideBar.props';
import useStyles from './SideBar.styles';

const SideBar: React.FC<SideBarProps> = () => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const theme = useTheme();

  const isXs = useMediaQuery(theme.breakpoints.down('xs'), { noSsr: true });
  const isSideBarOpen = useSelector(sideBarOpenedSelector);

  const handleSideBarClose = () => {
    dispatch(closeSideBar());
  };

  return (
    <Drawer
      variant={isXs ? 'temporary' : 'permanent'}
      open={isSideBarOpen}
      elevation={3}
      onClose={handleSideBarClose}
      className={classes.drawer}
      ModalProps={{
        disablePortal: true,
        keepMounted: true,
        BackdropProps: {
          className: classes.backdrop,
        },
      }}
      classes={{
        paper: classes.drawerPaper,
      }}
    >
      <Typography>SideBar</Typography>
    </Drawer>
  );
};

export default SideBar;
