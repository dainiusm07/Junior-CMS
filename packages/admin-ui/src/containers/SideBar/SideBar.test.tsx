import { render } from '@testing-library/react';

import { mockedUseSelector } from '../../redux/redux-mock-helpers';
import {
  mockMatchMedia,
  resetMatchMedia,
} from '../../test-utils/mock-match-media';
import theme from '../../theme';
import SideBar from './SideBar';

const mobileWidth = theme.breakpoints.values.xs - 1;
const desktopWidth = theme.breakpoints.values.xs;

describe('SideBar component', () => {
  afterEach(() => {
    resetMatchMedia();
  });

  [
    {
      description:
        'should match snapshot when in desktop view and SideBar is opened',
      width: desktopWidth,
      isOpened: true,
    },
    {
      description:
        'should match snapshot when in desktop view and SideBar is closed',
      width: desktopWidth,
      isOpened: false,
    },
    {
      description:
        'should match snapshot when in mobile view and SideBar is opened',
      width: mobileWidth,
      isOpened: true,
    },
    {
      description:
        'should match snapshot when in mobile view and SideBar is closed',
      width: mobileWidth,
      isOpened: false,
    },
  ].forEach(({ description, width, isOpened }) => {
    it(`${description}`, () => {
      mockMatchMedia(width);
      mockedUseSelector.mockReturnValueOnce(isOpened);
      const { asFragment } = render(<SideBar />, {});

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
