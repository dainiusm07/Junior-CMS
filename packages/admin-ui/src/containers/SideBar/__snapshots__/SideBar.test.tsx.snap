// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component should match snapshot when in desktop view and SideBar is closed 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        SideBar
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SideBar component should match snapshot when in desktop view and SideBar is opened 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        SideBar
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SideBar component should match snapshot when in mobile view and SideBar is closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="MuiDrawer-root MuiDrawer-modal makeStyles-drawer-10"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-12"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-11 MuiDrawer-paperAnchorLeft MuiPaper-elevation3"
      style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        SideBar
      </p>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;

exports[`SideBar component should match snapshot when in mobile view and SideBar is opened 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-modal makeStyles-drawer-7"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-9"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-8 MuiDrawer-paperAnchorLeft MuiPaper-elevation3"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        SideBar
      </p>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
